### PR TITLE
[libpq] Android compatibility

### PR DESCRIPTION
**Pull request by m-kuhn**
**Describe the pull request**

- #### What does your PR fix?  
  Fixes libpq cannot be linked on android < 26 because `nl_langinfo` isn't there.

I didn't find the proper condition to check for the required version. Improvements welcome.
Currently, 21 is the default on vcpkg and this version is broken.

Note that it's currently not possible to build libpq for android because the `make` integration for android is lacking. This patch is required anyways.

### CODE REVIEW

**Comment by Queens666** [comment]
#20430

**Comment by m-kuhn** [comment]
https://github.com/microsoft/vcpkg/issues/20844

File: ports/libpq/portfile.cmake [hunk 1]
```diff
@@ -278,6 +278,9 @@ else()
     else()
         list(APPEND BUILD_OPTS --without-readline)
     endif()
+    if(VCPKG_TARGET_IS_ANDROID) # AND CMAKE_SYSTEM_VERSION LESS 26)
+        list(APPEND BUILD_OPTS ac_cv_header_langinfo_h=no)
```
Comment (dg0yt) [inline ports/libpq/portfile.cmake:282]: Isn't this tested/autodetected in libpq's build system? You are just overriding the detected value IIUC.

Comment (Queens666) [reply ports/libpq/portfile.cmake:282]: -gracias

Comment (m-kuhn) [reply ports/libpq/portfile.cmake:282]: The header is there, the symbol not. That's why the detection is not appropriate.

**Review by NancyLi1013 — APPROVED** [review]

**Pull request merged**